tests/test_lattice.py

```diff
@@ -1748,3 +1997,530 @@ def test_layering_on_edge_cases():
     layers_single = get_compatible_layers(single_edge)
     assert layers_single == [[(0, 1)]]
     _validate_layers(single_edge, layers_single)
+
+
+def test_layering_on_disconnected_graph():
+    """
+    Tests that the layering algorithm correctly handles a graph consisting
+    of multiple disconnected components.
+    """
+    disconnected_bonds = [
+        (0, 1),
+        (1, 2),
+        (2, 0),
+        (3, 4),
+        (4, 5),
+        (5, 3),
+    ]
+
+    layers = get_compatible_layers(disconnected_bonds)
+
+    # Assert
+    _validate_layers(disconnected_bonds, layers)
+
+    assert len(layers) == 3, "Two disconnected triangles should be 3-edge-colorable."
+
+    layer_with_01 = next(
+        layer for layer in layers if (0, 1) in layer or (1, 0) in layer
+    )
+    assert (3, 4) in layer_with_01 or (4, 3) in layer_with_01
+
+
+# A map from backend name to the expected tensor type.
+BACKEND_TENSOR_MAP = {
+    "numpy": np.ndarray,
+    "jax": jnp.ndarray if jnp else None,
+    "tensorflow": tf.Tensor if tf else None,
+    "pytorch": torch.Tensor if torch else None,
+}
+
+
+class TestBackendIntegration:
+    """
+    Tests to ensure lattice functionalities are consistent and correct
+    across different computation backends.
+    """
+
+    # Define the test cases for parameterization
+    lattice_test_cases = [
+        (
+            SquareLattice,
+            {"size": (3, 3), "pbc": False},
+            (0, 4, np.sqrt(2.0)),
+            "SquareLattice",
+        ),
+        (
+            CustomizeLattice,
+            {
+                "dimensionality": 2,
+                "identifiers": [0, 1],
+                "coordinates": [[0.0, 0.0], [1.0, 1.0]],
+            },
+            (0, 1, np.sqrt(2.0)),
+            "CustomizeLattice",
+        ),
+        (
+            HoneycombLattice,
+            {"size": (2, 2), "pbc": True},
+            (0, 1, 1.0),
+            "HoneycombLattice",
+        ),
+        (
+            RectangularLattice,
+            {"size": (2, 3), "lattice_constants": (1.0, 2.0), "pbc": False},
+            (0, 1, 2.0),
+            "RectangularLattice",
+        ),
+    ]
+
+    @pytest.mark.parametrize("backend_name", ["numpy", "jax", "tensorflow", "pytorch"])
```

Comment:
use the backend fixture as in test_backends.py, also apply the backend fixture on all tests including the originally exisiting one to see their robustness across backends